frontend/public/module/k8s/selector-requirement.js

```diff
@@ -8,42 +8,40 @@ export const createEquals = (key, value) => ({
   values: [value],
 });
 
+const validateKey = (key, operator, values) => {
+  // Need to validate key after parts split or else we don't take correct branch
+  if (/^(.*[0-9A-Za-z])$/.test(key)) {
+    return { key, operator, values };
+  }
+};
```

Comment:
This method is a bit overloaded. It should only validate the key, and the callers should be responsible for returning the object.

```suggestion
const validateKey = (key) => /^(.*[0-9A-Za-z])$/.test(key);
```

Reply:
it is overloaded, but it seems much cleaner to do it this way then to either have:
```
if (validateKey(string)) { 
  return {key, 'Exists', []}
}
```
or 
```
validateKey(string) ? return {string, 'Exists' [] } : return;
```
That adds an extra conditional for 8 different branches.